### PR TITLE
Add "open xel file" support to profiler in sqltoolsservice

### PR DESCRIPTION
**Pull request by shueybubbles**
This update will enable ADS to implement XEL file open support. 
Key design tenets:

- An XEL file is treated as a regular XE session with a finite lifespan. When reading the file completes, the service sends a SessionStopped notification. 
- Don't break existing contracts for using ring buffer target.
- Lay the foundation for using the live target instead of the ring buffer for regular XE sessions in a future release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +37,7 @@ so that we can ensure we are solving the right problems in the right way.
 - Comment on existing issues to give your feedback on how they should be fixed/implemented
 - Contribute a bug fix or feature implementation by submitting a pull request
 - Contribute more unit tests for feature areas that lack good coverage
-- Review the pull requests that others submit to ensure they follow [established guidelines]
+ Review the pull requests that others submit to ensure they follow [established guidelines]
```
Comment (Charles-Gagnon) [inline README.md:40]: Why did you remove the - ?

File: src/Microsoft.SqlTools.ServiceLayer/Profiler/ObservableXEventSession.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            return sessionId;
+        }
+    }
+
+    /// <summary>
+    /// Source of ProfilerEvent push notifications. Wraps IXEventFetcher.
+    /// </summary>
+    public class XeStreamObservable : IObservable<ProfilerEvent>
+    {
+        private readonly object syncobj = new object();
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Profiler/ObservableXEventSession.cs:50]: nit: camel case would be nice here `syncObj`

**Comment by Charles-Gagnon** [comment]
Looks like you have a test failure as well https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=75409&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=0c9ca39c-92df-57a3-f66e-f5dca9672f46

**Comment by shueybubbles** [comment]
> Looks like you have a test failure as well https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=75409&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=0c9ca39c-92df-57a3-f66e-f5dca9672f46

hmm 
image

File: README.md (hunk 1 above)
Comment (shueybubbles) [reply README.md:40]: i think it had 2, i only meant to delete 1
BTW the markdownlint in vs code thinks our repo favors asterisks over hyphens for lists. Is that in a config file somewhere?
image

**Comment by Charles-Gagnon** [comment]
@shueybubbles Is this something you still plan on finishing at some point? If not I'd suggest closing this for now to clean up stale PRs

**Comment by shueybubbles** [comment]
i have no time to put into this and I bet the merge will be a mess
